GROUPS.md

```diff
@@ -0,0 +1,43 @@
+
+## Nested Groups
+
+To make more sophisticated scheduling constraints possible, it should be possible to tag a group itself with a group label.  Unfortunately "nesting" groups also invites the possibility of creating recursive groups.  "Group A" can be tagged with "Group B" and vice versa. It isn't clear what, if anything, should be done to prevent this other than to report any impossible scheduling constraints that result from nested groups.
+
+## Group Types
+
+Groups can be used to solve various problems in science program design.  In this section we attempt to enumerate the various options that will be needed.
+
+* _Organizational Folders._  These are arbitrary groups of observations with no impact on scheduling.  They can be used to organize observations that are logically similar or in some way important to the user.
+
+* _Any Groups._ This is an unordered group with a number (or possibly a percentage) parameter.  The idea is that *n* of the items in the group must be completed before the group is considered complete.  For example, this would cover the possibility of choosing amongst various options of identical value.
+
+  - Putting two items in a group and requiring just one of them to be done is effectively an "or" group.
+
+  - Setting n to the number of items in the group (or setting the percentage to 100 if that's the way these are done) would effectively result in an "and" group.
+
+* _Timing Groups._  We express timing constraints between two observations and/or groups with a timing group.  A timing group has a min/max time bounds specifying how much time must or may pass between execution of two consecutive items.
+
+  - It would be useful to extend this to *n* ordered items but it isn't clear to me how, beyond building up a hierarchy.
+
+  - Maybe this is too simplistic.  Are there other types of timing constraints besides min / max time bounds?
+
+  - Explicit ordering of items without any additional timing consideration could be represented via a timing group with no max time bounds and a min time bounds of 0.
+
+## Acquisitions and Telluric Standards
+
+Two common group use cases that have been mentioned in the past are telluric standard stars and acquisition observations. After discussion with Bryan and Andy we have concluded that neither warrants special group types.
+
+### Acquisitions
+
+The need for separate acquisition observations is a reflection of poor modeling choices in the existing codebase.  In particular they are necessary so that (differently configured) acquisition and science observations can be found by the OT browser which only takes into account the static instrument configuration.  In the new model, the acquisition sequence can just be included with the science sequence in the same observation and the need for separate observations and therefore a grouping of some kind disappears.
+
```

Comment:
We might note that steps *are* distinguished by time accounting category, so we can still tell which parts of a sequence are the acquisition and so on.